File: JobSearchApp.Core/Models/Vacancies/VacancyGetAllDto.cs
namespace JobSearchApp.Core.Models.Vacancies;

public class VacancyGetAllDto
{
    public int Id { get; set; }
    public string Title { get; set; }= null!;
    public string Description { get; set; }= null!;
    public double Salary { get; set; }
    public string Experience { get; set; }= null!;
    public string AttendanceMode { get; set; } = null!;
    public DateTime CreatedAt { get; set; }
    public string CompanyName { get; set; } = null!;
    public bool IsActive { get; set; }
    public IEnumerable<LocationDto> Locations { get; set; } = [];
}

File: client/src/models/vacancy/vacancy.getall.dto.ts
import { LocationDto } from "../common/location.dto.ts";

export interface VacancyGetAll {
    id: string;
    title: string;
    positionTitle: string;
    description: string;
    salary: number;
    experience: string;
    attendanceMode: string;
    createdAt: Date;
    companyName: string;
    locations: LocationDto[];
    isActive : boolean;
}